tests/testthat/test-connection.R

```diff
@@ -247,3 +252,27 @@ test_that("Error is thrown when forgetting password", {
   )
   expect_error(connection <- connect(details), "Connection propery 'password' is NULL")
 })
+
+
+test_that("dbms function maps DBI connections to correct SQL dialect", {
+  
+  mappings <- c(
+    'Microsoft SQL Server' = 'sql server',
+    'PqConnection' = 'postgresql',
+    'RedshiftConnection' = 'redshift',
+    'BigQueryConnection' = 'bigquery',
+    'SQLiteConnection' = 'sqlite',
+    'duckdb_connection'  = 'duckdb')
+  
+  
+  for(i in seq_along(mappings)) {
+    driver <- names(mappings)[i]
+    dialect <- unname(mappings)[i]
+    mockConstructor <- setClass(driver, contains = "DBIConnection")
```

Comment:
@schuemie I'm a little unsure of the call to setClass here. I want the effect of this call to be limited to this one test.